server/bundler/extensions-loader.js

```diff
@@ -0,0 +1,72 @@
+const fs = require( 'fs' );
+const path = require( 'path' );
+const camelCase = require( 'lodash' ).camelCase;
+const compact = require( 'lodash' ).compact;
+
+function getReducerPath( extensionDir ) {
+	return path.join( 'client', 'extensions', extensionDir, 'state', 'reducer.js' );
+}
+
+function generateReducerRequireString( extensionDir ) {
+	return `'${ camelCase( extensionDir ) }': require( 'extensions/${ extensionDir }/state/reducer' )`;
+}
+
+function generateExtensionsModuleString( reducerRequires ) {
+	return `module.exports = {
+		reducers: function() {
+			return {
+				${ reducerRequires.join( '\n' ) }
+			};
+		}
+	};`;
+}
+
+function getReducersSync( extensionDirs = [] ) {
+	return extensionDirs
+		.filter( ( extensionDir ) => fs.existsSync( getReducerPath( extensionDir ) ) )
+		.map( ( extensionDir ) => generateReducerRequireString( extensionDir ) );
```

Comment:
The above could be just `.map( generateReducerRequireString );`, but totally not a blocker.